modules/querying/pages/query-operations.adoc

```diff
@@ -577,7 +577,17 @@ Installing takes several seconds for each query. The current version does not su
 Concurrent `INSTALL QUERY` commands are allowed as long as only one `INSTALL QUERY` command is running on a single graph. Concurrent `INSTALL QUERY` commands are _not_ allowed on a single graph.
 ====
 
-=== Options for `INSTALL QUERY`
+=== Query optimizer (preview)
+When installing a query, you have the option of using the query optimizer.
+The query optimizer can potentially reduce the amount of memory a query requires and the time it takes to finish execution.
+The optimizer works by automatically finds the most efficient traversal plan when compiling the query during query installation.
+You can benefit from this option if you query uses a `FROM` clause that performs any number of hops.
+
+To use the query optimizer, use the `-COST` option when installing a query.
+This option can be used together with other options, but it must come after the other options.
+For example, you cannot run `INSTALL QUERY ALL -COST -FORCE`, it must be `INSTALL QUERY ALL -FORCE -COST`
+
+=== Options
 
```

Comment:
```can potentially reduce the amount of memory a query```

We do not optimize to reduce memory consumption (not with this release).

Reply:
Here it is not clear that this is a cost-based query optimizer.

Cost-based optimizer needs a statistics to have accurate estimation in order to find the good plans.

I think query optimizer needs it's own page (or sub-page), with more info. To explain that there is also a session variable to enable optimizer. To talk about statistics gathering and how to do it. Similarly to how conjunctive patterns, per clause, and etc advanced features have more of a dedicated pages. 


And here for install option we can keep what you wrote, but also refer user for a cost opt page for full info.

Reply:
Regarding stats: Stats has cardinality (vertex, edge counts) and histograms.  i'd recommend for users to collect refined edge counts (at least) and build histograms on attributes that will be used in WHERE conditions.

Reply:
and by "dedicated page" and "full info" i just meant: all the necessary info to start using cost-optimizer.

Reply:
Hi Ainur, thanks for the comment. I spoke with Joshua and will add to the doc to reflect the whole process of how to use the optimizer. It was not clear to me before in the QA doc.